src/recipe/recipe.controller.spec.ts

```diff
@@ -1,18 +1,175 @@
 
-describe('RecipeController', () => {
-  let controller: RecipeController;
+describe("Recipe Controller's tests", () => {
+    let recipeController: RecipeController;
+    let recipeService: RecipeService;
+    let prismaService: PrismaService;
 
-  beforeEach(async () => {
-    const module: TestingModule = await Test.createTestingModule({
-      controllers: [RecipeController],
-    }).compile();
+    const mockRecipe = {
+        id: expect.any(Number),
+        createdAt: expect.any(Number),
+        title: expect.any(String),
+        description: expect.any(String),
+        ingredients: expect.any(String),
+        preparation: expect.any(String),
+    };
 
-    controller = module.get<RecipeController>(RecipeController);
-  });
+    class mockRecipeRequest {
+        title: string;
+        description: string;
+        ingredients: string;
+        preparation: string;
+    }
 
-  it('should be defined', () => {
-    expect(controller).toBeDefined();
-  });
+    const mockRecipeService = {
+        createRecipe: jest.fn((dto: mockRecipeRequest) => {
+            return {
+                id: 1,
+                createdAt: Date.now(),
+                ...dto,
+            };
+        }),
+        fetchRecipe: jest.fn((id: number) => {
+            return {
+                id,
+                createdAt: Date.now(),
+                title: 'Fetched recipe title',
+                description: 'Fetched recipe description',
+                ingredients: 'Fetched recipe ingredients',
+                preparation: 'Fetched recipe preparation',
+            };
+        }),
+        fetchAllRecipes: jest.fn(() => {
+            return [
+                {
+                    id: 0,
+                    createdAt: Date.now(),
+                    title: 'Fetched recipe title',
+                    description: 'Fetched recipe description',
+                    ingredients: 'Fetched recipe ingredients',
+                    preparation: 'Fetched recipe preparation',
+                },
+                {
+                    id: 1,
+                    createdAt: Date.now(),
+                    title: 'Fetched recipe title',
+                    description: 'Fetched recipe description',
+                    ingredients: 'Fetched recipe ingredients',
+                    preparation: 'Fetched recipe preparation',
+                },
+            ];
+        }),
+        updateRecipe: jest.fn((id: number, dto: mockRecipeRequest) => {
+            return {
+                id,
+                createdAt: Date.now(),
+                title: dto.title || 'latest title',
+                description: dto.description || 'latest description',
+                ingredients: dto.ingredients || 'latest ingredients',
+                preparation: dto.preparation || 'latest preparation',
+            };
+        }),
+        deleteRecipe: jest.fn((_id: number) => {
+            return Promise<void>
```

Comment:
hmm this look kind of strange - `Promise<void>` is a type, not something we return. If it works then ok, but perhaps this could be better:
```typescript
deleteRecipe: jest.fn()
```

or:
```typescript
deleteRecipe: jest.fn((id: number) => {
  return Promise.resolve();
})
```

or:
```typescript
deleteRecipe: jest.fn((id: number) => Promise.resolve())
```